tests/h/services/nipsa_test.py

```diff
@@ -80,44 +64,38 @@ def test_fetch_all_flagged_userids_caches_lookup(self, db_session, users):
             "acct:cecilia@example.com",
         }
 
-    def test_flag_updates_cache(self, db_session, users):
-        svc = NipsaService(db_session)
-
+    def test_flag_updates_cache(self, svc, users):
         svc.fetch_all_flagged_userids()
         svc.flag(users["dominic"])
         users["dominic"].nipsa = False  # Make sure result below comes from cache.
 
         assert svc.is_flagged(users["dominic"].userid)
 
-    def test_unflag_updates_cache(self, db_session, users):
-        svc = NipsaService(db_session)
-
+    def test_unflag_updates_cache(self, svc, users):
         svc.fetch_all_flagged_userids()
         svc.unflag(users["renata"])
         users["renata"].nipsa = True  # Make sure result below comes from cache.
 
         assert not svc.is_flagged(users["renata"].userid)
 
-    def test_clear_resets_cache(self, db_session, users):
-        svc = NipsaService(db_session)
-
+    def test_clear_resets_cache(self, svc, users):
         svc.fetch_all_flagged_userids()
         users["renata"].nipsa = False
         svc.clear()
 
         assert not svc.is_flagged("acct:renata@example.com")
 
+    @pytest.fixture
+    def svc(self, db_session, search_index):
+        return NipsaService(db_session, search_index)
 
-def test_nipsa_factory(pyramid_request):
+
+def test_nipsa_factory(pyramid_request, search_index):
     svc = nipsa_factory(None, pyramid_request)
 
     assert isinstance(svc, NipsaService)
     assert svc.session == pyramid_request.db
-
-
-@pytest.fixture
-def reindex_user_annotations(patch):
-    return patch("h.services.nipsa.reindex_user_annotations")
+    assert svc._search_index == search_index
 
 
```

Comment:
Again, nothing you did, but the `users` fixture defined here is specific to the TestNipsaService and then included with `@pytest.mark.usefixtures("users")`. We could just drop it into the end of the class with `autouse=True` and it wouldn't look like it applied to the `test_nipsa_factory` as well.

Reply:
Yeah that belongs in the class. I quite like `usefixtures` on classes (which you can still do even if the fixture is defined in the class) as it declares, at the top of the class, which fixtures are always used. But I think stopping people from using the handy `autouse=True` will be impossible